chain/client/src/client_actor.rs

```diff
@@ -230,22 +229,21 @@ impl Handler<NetworkClientMessages> for ClientActor {
                 }
             }
             NetworkClientMessages::StateRequest(shard_id, hash, need_header, parts, route_back) => {
-                let network_adapter = self.network_adapter.clone();
                 // TODO discuss vulnerability: a node can be flooded by StateRequests
-                actix::spawn(
-                    self.client
-                        .chain
-                        .get_state_response_by_request(shard_id, hash, need_header, parts)
-                        .then(move |result| {
-                            if let Ok(shard_state) = result {
-                                network_adapter.do_send(NetworkRequests::StateResponse {
-                                    response: StateResponseInfo { shard_id, hash, shard_state },
-                                    route_back,
-                                });
-                            }
-                            future::ready(())
-                        }),
-                );
+                ctx.run_later(Duration::from_millis(0), move |act, _ctx| {
```

Comment:
Why is this better than `actix::spawn`?

Reply:
AFAIU in the current implementation the function get_state_response_by_request is blocking and don't allow fast response. Ideally it should be something like wrapping this is called inside future::lazy, but you don't have access to self there, and this way it is guaranteed by actix.

Reply:
To be more precise get_state_response_by_request should be async